lms/models/grading_sync.py

```diff
@@ -0,0 +1,60 @@
+from enum import StrEnum
+from typing import TYPE_CHECKING
+
+from sqlalchemy import ForeignKey, text
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from lms.db import Base, varchar_enum
+from lms.models import Assignment
+from lms.models._mixins import CreatedUpdatedMixin
+
+if TYPE_CHECKING:
+    from lms.models import LMSUser
+
+
+class AutoGradingSyncStatus(StrEnum):
+    SCHEDULED = "scheduled"
+    IN_PROGRESS = "in_progress"
+    FINISHED = "finished"
+    FAILED = "failed"
+
+
+class GradingSync(CreatedUpdatedMixin, Base):
+    __tablename__ = "grading_sync"
+
+    id: Mapped[int] = mapped_column(autoincrement=True, primary_key=True)
+
+    assignment_id: Mapped[int] = mapped_column(ForeignKey("assignment.id"), index=True)
+    assignment: Mapped[Assignment] = relationship()
+
+    status: Mapped[str | None] = varchar_enum(AutoGradingSyncStatus)
+
+    created_by_id: Mapped[int] = mapped_column(ForeignKey("lms_user.id"))
+    created_by: Mapped["LMSUser"] = relationship()
+    """Who created this grade sync."""
+
+
+class GradingSyncGrade(CreatedUpdatedMixin, Base):
+    __tablename__ = "grading_sync_grade"
+
+    id: Mapped[int] = mapped_column(autoincrement=True, primary_key=True)
+
+    grading_sync_id: Mapped[int] = mapped_column(
+        ForeignKey("grading_sync.id"), index=True
+    )
+    grading_sync: Mapped[GradingSync] = relationship(backref="grades")
+
+    lms_user_id: Mapped[int] = mapped_column(ForeignKey("lms_user.id"), index=True)
+    lms_user: Mapped["LMSUser"] = relationship()
+    """Who this grade belongs to"""
+
+    grade: Mapped[float] = mapped_column()
+
+    error_details: Mapped[JSONB] = mapped_column(
+        JSONB(), server_default=text("'{}'::jsonb"), nullable=False
+    )
+    """Any extra information about potential errors while syncing this grade to the LMS"""
+
+    success: Mapped[bool | None] = mapped_column()
```

Comment:
This as a boolean in the DB. We can later expose it as `status` at the API level.